erigon-lib/diagnostics/stages.go

```diff
@@ -166,19 +168,11 @@ func (d *DiagnosticClient) runSubStageListener(rootCtx context.Context) {
 				d.mu.Lock()
 				d.SetSubStagesList(info.Stage, info.List)
 				d.mu.Unlock()
-
-				d.saveSyncStagesToDB()
 			}
 		}
 	}()
 }
 
-func (d *DiagnosticClient) saveSyncStagesToDB() {
```

Comment:
When using approach with flushing to database by timeout can easily forget to flush once it's really needed (e.g. When stage is over)

Is flushing once in 5 min is really what you wanted?

Reply:
done